pkg/routes/gacha-routes.go

```diff
@@ -3,13 +3,15 @@ package routes
 import (
 	"github.com/gorilla/mux"
 	"github.com/iamananya/ginco-task/pkg/controllers"
+	"github.com/iamananya/ginco-task/pkg/middlewares"
 )
 
 var RegisterUserRoute = func(router *mux.Router) {
 	router.HandleFunc("/user/", controllers.CreateUser).Methods("POST")
-	router.HandleFunc("/user/", controllers.GetUser).Methods("GET")
-	router.HandleFunc("/user/{userId}", controllers.GetUserById).Methods("GET")
-	router.HandleFunc("/user/{userId}", controllers.UpdateUser).Methods("PUT")
-	// router.HandleFunc("/user/{userId}", controllers.DeleteUser).Methods("DELETE")
+	router.HandleFunc("/user/", middlewares.AuthenticationMiddleware(controllers.GetUser)).Methods("GET")
+	router.HandleFunc("/user/", middlewares.AuthenticationMiddleware(controllers.UpdateUser)).Methods("PUT")
 
+	router.HandleFunc("/characters/list/", controllers.ListCharacters).Methods("GET")
+	router.HandleFunc("/gacha/draw/", controllers.HandleGachaDraw).Methods("POST")
```

Comment:
There should be authentication for ListCharacters and DrawGacha.

Reply:
@GuiltyMorishita Okay,I will implement this in a separate PR.